Add documentation for version 2 (#6)
laminas/laminas-mvc-middleware: docs/book/v2/dispatching-middleware.md

    ],
];
```

##  Anonymous Interface Implementations

You can also pass an anonymous class which implements `MiddlewareInterface` or `RequestHandlerInterface` directly.
This is intended primarily for development because it can cause issues with configuration caching.

```php
use Laminas\Diactoros\Response\TextResponse;
use Laminas\Mvc\Middleware\PipeSpec;
use Laminas\Router\Http\Literal;
use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface;
use Psr\Http\Server\MiddlewareInterface;
use Psr\Http\Server\RequestHandlerInterface;

return [
    'router' => [
        'routes' => [
            'instance-middleware' => [
                'type' => Literal::class,
                'options' => [
                    'route' => '/instance-middleware',
                    'defaults' => [
                        'controller' => PipeSpec::class,
                        'middleware' => new class implements MiddlewareInterface {
                            public function process(
                                ServerRequestInterface $request,
                                RequestHandlerInterface $handler
                            ): ResponseInterface {

>                                 // You have to return a response here because attempting to use $handler here will

Reviewer: Double "here" seems redundant?
```suggestion
                                // You have to return a response directly because attempting to use $handler here will
```

                    'defaults' => [
                        'controller' => PipeSpec::class,
                        'middleware' => new PipeSpec(

                            // id/alias of service in container
                            'container_id_for_album_middleware',

                            // anonymous class implementing a MiddlewareInterface
                            new class implements MiddlewareInterface {
                                public function process(
                                    ServerRequestInterface $request,
                                    RequestHandlerInterface $handler
                                ): ResponseInterface {
                                    return $handler->handle($request);
                                }
                            },

                            // Closure
                            static function (
                                ServerRequestInterface $request,
                                RequestHandlerInterface $handler
                            ): ResponseInterface {
                                return $handler->handle($request);
                            },

>                             // class-string of an Handler which will be resolved by the container

Reviewer: This might imply that such usage will use auto-invokable feature or some kind of abstract factory. No such feature is provided however.
This might be a more clear way to describe the usage: 
```suggestion
                            // class-string of a handler as a convenient way to specify id/alias which must be resolved by the container
```

                            AlbumDetailHandler::class,
                        ),
                    ],
                ],
            ],
        ],
    ],
];
```

> ### Note
>

> > Please note that if you pass instances of middleware or handlers directly, it is not compatible with config caching. You
> > should use that for development only. The recommended variant is passing container managed middleware
> > like `SomeRequestHandler::class`.

Reviewer: Feature to serialized objects and closures was added to config aggregator and service manager, so this should be slightly adjusted
```suggestion
> Please note that if you pass instances of middleware or handlers directly, it might not be compatible with config caching. You
> should use that for development only. The recommended variant is passing container managed middleware
> like `SomeRequestHandler::class`.
```
